recipes/mysql-connector-c/all/conanfile.py

```diff
@@ -20,10 +20,10 @@ class MysqlConnectorCConan(ConanFile):
 
     def requirements(self):
         if self.options.with_ssl:
-            self.requires.add("openssl/1.0.2t")
+            self.requires.add("openssl/1.0.2t", private=self.options.shared)
```

Comment:
This is broken if the transitive dependency is a shared library too, which is possible, actually very probable, if some user specifies ``openssl:shared=True``. Private dependencies can and should only be used in very specific situations, but unfortunately this is not one that can be guaranteed.

We have discussed many times, and we have plans to improve the dependency graph model and package relations, so these things can be better represented. In the meantime, I think it is better to go with regular dependencies, I totally understand the use cases this is trying to improve, but still some of them have counter arguments, for example:

> The consumer can use versions of OpenSSL and zlib libraries which are not compatible with dependencies for conan-mysql-connector-c.

Yes, but also if you make it private, it is not possible to update the dependency on OpenSSL anymore, and even if the consumer would like mysql-connector-c to use an updated openssl, that will not be possible without going to the recipe, forking it and changing it, because private dependencies cannot be overridden from downstream consumers.